benchmarks/src/jmh/java/software/amazon/jdbc/benchmarks/ReadWriteSplittingConnectionPoolingBenchmarks.java

```diff
@@ -0,0 +1,165 @@
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package software.amazon.jdbc.benchmarks;
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+import com.zaxxer.hikari.HikariConfig;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Timeout;
+import org.openjdk.jmh.annotations.Warmup;
+import org.postgresql.PGProperty;
+import software.amazon.jdbc.ConnectionProviderManager;
+import software.amazon.jdbc.Driver;
+import software.amazon.jdbc.HikariPooledConnectionProvider;
+import software.amazon.jdbc.PropertyDefinition;
+
+@State(Scope.Benchmark)
+@Fork(1)
+@Warmup(iterations = 1)
+@Timeout(time = 60, timeUnit = TimeUnit.MINUTES)
+@Measurement(iterations = 1)
+@BenchmarkMode({Mode.SingleShotTime})
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+public class ReadWriteSplittingConnectionPoolingBenchmarks {
+
+  // User configures connection properties here
+  private static final String POSTGRESQL_CONNECTION_STRING =
+      "jdbc:aws-wrapper:postgresql://database-pg-name.cluster-XYZ.us-east-2.rds.amazonaws.com:5432/connectionSample";
+  private static final String USERNAME = "username";
+  private static final String PASSWORD = "password";
+
+  private static final int NUM_ITERATIONS = 10;
+  private static final int NUM_THREADS = 10;
+  private static final String QUERY_1 = "SELECT pg_sleep(10)";
+
+  @Setup(Level.Iteration)
+  public static void setUp() throws SQLException {
+    if (!org.postgresql.Driver.isRegistered()) {
+      org.postgresql.Driver.register();
+    }
+
+    if (!Driver.isRegistered()) {
+      Driver.register();
+    }
+  }
+
+  protected Connection connectToInstance(final String url, final Properties props)
+      throws SQLException {
+    return DriverManager.getConnection(url, props);
+  }
+
+  protected Properties initNoPluginPropsWithTimeouts() {
```

Comment:
nit: the method name mentions there are timeouts but there aren't any in the implementation. We should either change the name or add timeouts. Since the previous version of this code did not have timeouts here either I'm guessing they are not needed

Reply:
thanks!